### PR TITLE
[FIX] fix type requirements of operator().

### PR DESCRIPTION
**Pull request by smehringer**

### CODE REVIEW

File: include/seqan3/argument_parser/validators.hpp [hunk 1]
```diff
@@ -200,7 +200,7 @@ class value_list_validator
      */
     template <std::ranges::ForwardRange range_type>
     //!\cond
-        requires std::CommonReference<value_type_t<range_type>, value_type>
+        requires ImplicitlyConvertibleTo<value_type_t<range_type>, value_type const &>
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: Is there a specific reason not to use std::ConvertibleTo?

File: include/seqan3/argument_parser/validators.hpp [hunk 2]
```diff
@@ -289,7 +289,7 @@ class file_ext_validator
      */
     template <std::ranges::ForwardRange range_type>
     //!\cond
-        requires std::CommonReference<value_type_t<range_type>, std::filesystem::path>
+        requires ImplicitlyConvertibleTo<value_type_t<range_type>, std::filesystem::path const &>
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: CommonReference in documentation

File: include/seqan3/argument_parser/validators.hpp [hunk 3]
```diff
@@ -341,7 +341,7 @@ class file_existance_validator
...
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: common reference in doc

File: include/seqan3/argument_parser/validators.hpp [hunk 4]
```diff
@@ -404,7 +404,7 @@ class regex_validator
      */
     template <std::ranges::ForwardRange range_type>
     //!\cond
-        requires std::CommonReference<value_type_t<range_type>, value_type>
+        requires ImplicitlyConvertibleTo<value_type_t<range_type>, value_type const &>
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: std::ConvertibleTo and doc

File: include/seqan3/argument_parser/validators.hpp [hunk 5]
```diff
@@ -495,13 +495,18 @@ class validator_chain_adaptor
...
+     * \param[in] cmp   The value to validate.
      *
      * This function delegates to the validation of both of the chained validators
      * by calling their operator() one after the other. The behaviour depends on
      * the chained validators which may throw on input error.
      */
-    void operator()(value_type const & cmp) const
+    template <typename cmp_type>
+    //!\cond
+        requires std::Invocable<validator1_type, cmp_type> && std::Invocable<validator2_type, cmp_type>
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]:
```suggestion
        requires std::Invocable<validator1_type, cmp_type const> && std::Invocable<validator2_type, cmp_type const>
```

**Comment by smehringer** [comment]
@rrahn Review ping

**Comment by smehringer** [comment]
@rrahn review ping

**Pull request merged**